src/arcaflow_plugin_sdk/schema.py

```diff
@@ -5991,14 +5991,15 @@ def __call__(
         result = self._handler(step_local_data.initialized_object, params)
         if not isinstance(result, tuple):
             raise BadArgumentException(
-                "The step returned type {};".format(type(result))
-                + "  expected a tuple with two values: output ID string"
-                  " and a step-specific value."
+                f"The implementation of step {run_id}/{self.id} returned"
+                f" type {type(result)}; expected a tuple with two"
+                " values: output ID string and a step-specific value."
             )
         if len(result) != 2:
             raise BadArgumentException(
-                "The step returned {} results instead of 2. Did your step"
-                " return the correct results?".format(len(result))
+                f"The implementation of step {run_id}/{self.id} returned"
+                f"{len(result)} results instead of 2. Did your step"
+                " return the correct results?"
```

Comment:
Including the step identification is excellent; however, asking if it returned the correct results, _after having pointed out that it returned the wrong number_ is a bit...salty?  If you really want to be helpful, you could consider including the value of the result in addition to or instead of the length.  (Ditto [what Dave said](https://github.com/arcalot/arcaflow-plugin-sdk-python/pull/134#discussion_r1651510882). 😀)

Reply:
Sure.